Solution/WorkingFilesList.ToolWindow.Test/ViewModel/DocumentMetadataManagerTests.cs

```diff
@@ -2294,5 +2294,236 @@ public void ClearRemovesAllMetadata()
             Assert.IsEmpty(manager.PinnedDocumentMetadata);
             Assert.IsEmpty(collection);
         }
+
+        public IDocumentMetadataManager PrepareDocumentsForFilterTest(params string[] docNames)
+        {
+            var documentMockList = docNames.Select(d => CreateDocumentWithInfo(new DocumentMetadataInfo
+            {
+                FullName = d,
+                ProjectDisplayName = d
+            })).ToList();
+
+            var documents = CreateDocuments(documentMockList);
+
+            var metadataFactoryBuilder = new DocumentMetadataFactoryBuilder();
+            metadataFactoryBuilder.FilePathServiceMock
+                .Setup(f => f.ReducePath(
+                    It.IsAny<string>(),
+                    It.IsAny<int>()))
+                .Returns<string, int>((str, i) => str);
+
+            metadataFactoryBuilder.DisplayNameHighlightEvaluatorMock
+                .Setup(d => d.GetHighlight(
+                    It.IsAny<string>(),
+                    It.IsAny<bool>()))
+                .Returns<string, bool>((str, b) => str);
+
+            var factory = metadataFactoryBuilder.CreateDocumentMetadataFactory(true);
+
+            var builder = new DocumentMetadataManagerBuilder
+            {
+                DocumentMetadataFactory = factory
+            };
+
+            var manager = builder.CreateDocumentMetadataManager();
+            manager.Synchronize(documents, false);
+
+            foreach (var d in (IList<DocumentMetadata>)manager.ActiveDocumentMetadata.SourceCollection)
+            {
+                manager.TogglePinnedStatus(d);
+            }
+            manager.ActiveDocumentMetadata.Refresh();
+
+            return manager;
+        }
+
+        [Test]
+        public void FilterMatchesTwoDocuments()
+        {
+            // Arrange
+            const string document1Name = "AAADocument1";
+            const string document2Name = "BBBDocument2";
+            const string document3Name = "AABDocument3";
+
```

Comment:
As all of the tests use three or four document names that are fairly similar, I’d suggest pulling these out as `const string`s into the new inner class; something like
```
private const string AaaDocument1 = "AAADocument1";
```